### PR TITLE
Fyle 1rwz3mg : Sent Back Task for Individual Advances

### PR DESCRIPTION
**Pull request by siddharth0-0**
https://user-images.githubusercontent.com/66034160/153176030-f000e3ca-d52c-4e53-b837-c6aa877d4d12.mp4

### CODE REVIEW

**Comment by Dimple16** [comment]
fix lint issues

File: src/app/core/services/tasks.service.ts [hunk 1]
```diff
@@ -127,6 +135,13 @@ export class TasksService {
       }
     }
 
+    if (filters.sentBackAdvances) {
+      const existingFilter = selectedFilters.find((filter) => filter.name === 'Advances');
+      if (existingFilter) {
+        existingFilter.value.push('SENT_BACK');
+      }
```
Comment (Dimple16) [inline src/app/core/services/tasks.service.ts:142]: else? will this have any default?

File: src/app/core/services/tasks.service.ts [hunk 2]
```diff
@@ -443,6 +514,34 @@ export class TasksService {
     }
   }
 
+  mapSentBackAdvacnesToTasks(
```
Comment (Dimple16) [inline src/app/core/services/tasks.service.ts:517]: `mapSentBackAdvancesToTasks` typo

**Review by Dimple16 — CHANGES_REQUESTED** [review]
add trackers wherever required

**Comment by Dimple16** [comment]
trackers aren't added yet, please check

**Review by Dimple16 — APPROVED** [review]

File: src/app/core/services/tasks.service.ts [hunk 3]
```diff
@@ -104,39 +104,53 @@ export class TasksService {
...
+      this.getSelectedFilter(selectedFilters, 'Reports', 'TEAM');
     }
 
+    if (filters.sentBackAdvances) {
+      this.getSelectedFilter(selectedFilters, 'Advances', 'SENT_BACK');
+    }
     return selectedFilters;
   }
 
+  getSelectedFilter(selectedFilters, filterName, value) {
```
Comment (Julias0) [inline src/app/core/services/tasks.service.ts:120]: define datatype of the arguments passed

also rename this method - I dont understand what this function does by reading its name

Comment (Julias0) [reply src/app/core/services/tasks.service.ts:120]: Also make this a proper pure function - @Dimple16 can help here

Comment (siddharth0-0) [reply src/app/core/services/tasks.service.ts]: Have removed the function itself as it was no longer needed

File: src/app/core/services/advance.service.ts [hunk 4]
```diff
@@ -91,4 +91,18 @@ export class AdvanceService {
     }
     return data;
   }
+
+  getAdvancesStats(params) {
```
Comment (Julias0) [inline src/app/core/services/advance.service.ts:95]: getAdvancesStats -> getMyAdvanceRequestStats

Comment (Julias0) [reply src/app/core/services/advance.service.ts:95]: params should have a datatype

Comment (Julias0) [reply src/app/core/services/advance.service.ts:95]: add a return type

File: src/app/core/services/advance.service.ts [hunk 5]
```diff
@@ -91,4 +91,18 @@ export class AdvanceService {
...
+    return from(this.authService.getEou()).pipe(
+      switchMap((eou) =>
+        this.apiv2Service.get('/advance_requests/stats', {
```
Comment (Julias0) [inline src/app/core/services/advance.service.ts:98]: extract out into a separate function

File: src/app/core/services/tasks.service.ts [hunk 6]
```diff
@@ -443,6 +514,34 @@ export class TasksService {
...
+  private mapSentBackAdvancesToTasks(
+    aggregate: { totalCount: number; totalAmount: number },
+    homeCurrency: string
+  ): DashboardTask[] {
+    if (aggregate.totalCount > 0) {
+      return [
+        {
+          amount: this.humanizeCurrency.transform(aggregate.totalAmount, homeCurrency, 2, true),
+          count: aggregate.totalCount,
+          header: `Advance${aggregate.totalCount === 1 ? '' : 's'} sent back!`,
```
Comment (Julias0) [inline src/app/core/services/tasks.service.ts:526]: do these in separate variables and use the variables here - it is hard to read like this
Example - 
```typescript
header: `Advance${aggregate.totalCount === 1 ? '' : 's'} sent back!`, -> 

const header = `Advance${aggregate.totalCount === 1 ? '' : 's'} sent back!`;

{
...
header,
...
}
```

File: src/app/fyle/dashboard/tasks/tasks.component.ts [hunk 7]
```diff
@@ -370,6 +407,30 @@ export class TasksComponent implements OnInit {
...
   }
 
+  onSentBackAdvanceTaskClick(taskCta: TaskCta, task: DashboardTask) {
+    if (task.count === 1) {
+      const queryParams = {
+        areq_state: 'in.(DRAFT)',
+        areq_is_sent_back: 'is.true',
+      };
+
+      from(this.loaderService.showLoader('Opening your advance...'))
```
Comment (Julias0) [inline src/app/fyle/dashboard/tasks/tasks.component.ts:417]: Opening your advance.. -> Opening your advance request

Note: Advance and advance requests are separate things, make sure you dont mix them up

File: src/app/core/services/advance.service.ts (hunk 4 above)
Comment (siddharth0-0) [reply src/app/core/services/advance.service.ts]: moved function in `advance-request.service.ts`

**Review by Julias0 — APPROVED** [review]

**Pull request merged**